core/camel-core-reifier/src/main/java/org/apache/camel/reifier/TransformReifier.java

```diff
@@ -31,6 +32,10 @@ public TransformReifier(Route route, ProcessorDefinition<?> definition) {
 
     @Override
     public Processor createProcessor() throws Exception {
+        if (definition.getToType() != null) {
+            return new DataTypeProcessor(definition.getFromType(), definition.getToType());
```

Comment:
Hmm I think we may need to throw an illegal argument exception if you have configured toType and also expression. It seems that this is not possible.

Reply:
yes 👍